application/src/main/java/org/togetherjava/tjbot/commands/help/BookmarkPaginatorInteractor.java

```diff
@@ -0,0 +1,479 @@
+                this::onRemovePaginationRequest);
+    }
+
+    @Override
+    public String getName() {
+        return "bookmarks-paginator";
+    }
+
+    @Override
+    public UserInteractionType getInteractionType() {
+        return UserInteractionType.OTHER;
+    }
+
+    @Override
+    public void acceptComponentIdGenerator(ComponentIdGenerator generator) {
+        componentIdInteractor.acceptComponentIdGenerator(generator);
+    }
+
+    /**
+     * Generates a component id with the arguments the {@link Arraylizable} class provides.
+     *
+     * @param args The {@link Arraylizable} class to get the arguments from
+     * @return The generated component ID
+     */
+    private String generateComponentId(Arraylizable<String> args) {
+        return componentIdInteractor.generateComponentId(args.toArray());
+    }
+
+    @Override
+    public void onModalSubmitted(ModalInteractionEvent event, List<String> stringArgs) {
+        throw new UnsupportedOperationException("Not used");
+    }
+
+    @Override
+    public void onButtonClick(ButtonInteractionEvent event, List<String> stringArgs) {
+        ComponentArguments args = parseStringArgs(stringArgs);
+        long userID = event.getUser().getIdLong();
+        List<BookmarksRecord> bookmarks = bookmarksSystem.getUsersBookmarks(userID);
+
+        args.currentPageIndex = switch (args.componentName) {
+            case BUTTON_PREV_NAME -> ensurePageIndex(bookmarks, args.currentPageIndex - 1);
+            case BUTTON_NEXT_NAME -> ensurePageIndex(bookmarks, args.currentPageIndex + 1);
+            case BUTTON_REMOVE_NAME -> {
+                RemoveComponentArguments removeArgs = (RemoveComponentArguments) args;
+
+                removeSelectedBookmarks(bookmarks, event, removeArgs);
+
+                yield ensurePageIndex(bookmarks, args.currentPageIndex);
+            }
+            default -> throw new IllegalArgumentException("Unknown button");
+        };
+
+        updatePagination(event, args, bookmarks);
+    }
+
+    @Override
+    public void onSelectMenuSelection(SelectMenuInteractionEvent event, List<String> stringArgs) {
+        List<SelectOption> selected = event.getSelectedOptions();
+        ComponentArguments args = parseStringArgs(stringArgs);
+        long userID = event.getUser().getIdLong();
+        List<BookmarksRecord> bookmarks = bookmarksSystem.getUsersBookmarks(userID);
+
+        if (args.paginationType == PaginationType.REMOVE) {
+            RemoveComponentArguments removeArgs = (RemoveComponentArguments) args;
+
+            if (removeArgs.componentName.equals(SELECTMENU_REMOVE_NAME)) {
+                List<Long> bookmarksToRemoveChannelIDs =
+                        selected.stream().map(o -> Long.parseLong(o.getValue())).toList();
+
+                args = new RemoveComponentArguments(args.paginationType, args.componentName,
+                        args.currentPageIndex, bookmarksToRemoveChannelIDs);
```

Comment:
Dont overwrite args and just overwrite bookmarksToRemoveChannelIDs
```suggestion
                removeArgs.bookmarksToRemoveChannelIDs = bookmarksToRemoveChannelIDs;
```